### DISCUSSION TITLE
Need guide to add Bengali language

### DISCUSSION THREAD

**Discussion by Iftakharpy** [monkeytypegame/monkeytype#1517] [Questions] [answered]
# I want to contribute to this project by adding the Bengali language to the typing tests.

I haven't contributed to any open-source project before but I am confident that I would be able to do it but don't know where to start. And what's the best way to contribute to saving both my and your time.

Hoping for your advice.

**Comment by marie-bnl** [comment, marked as answer]
You'll need a list of the 200 most used Bengali words

You can find one here : https://raw.githubusercontent.com/hermitdave/FrequencyWords/master/content/2018/bn/bn_full.txt  
(so take only the 200 first lines)

Then you need to create a `bengali.json` file in `static/languages/` (take `english.json` as an example)

Then reference your newly created file in `_list.json` and `_groups.json`

See this for an example where I added Esperanto language  
https://github.com/Miodec/monkeytype/pull/1515/files

**Reply by Iftakharpy** [reply]
@victorbnl, I am getting an error when I run `npm install`. 
![Capture](https://user-images.githubusercontent.com/37600593/120913633-493ec800-c6ba-11eb-80b8-02c6a3a34513.PNG)

Can you help me fix the error?

**Reply by Otard95** [reply]
Its trying to install `node-sass` which if you check out their [install guide](https://github.com/sass/node-sass#install) requires [`node-gyp`](https://github.com/nodejs/node-gyp#on-windows), that might be a good place to start.

Edit: this might also help - https://stackoverflow.com/a/54059596  
Also its using python 3 some comments say it might need python 2.

**Reply by marie-bnl** [reply]
I dont know if that counts as a solution or just a hacky wordaround but with NodeJS 14 it works fine, not with 15 and 16

**Reply by Iftakharpy** [reply]
When I installed firebase-tools `functions/package-lock.json` and `package-lock.json` files were modified. Should I add these changes to my PR?

**Reply by marie-bnl** [reply]
No you should'nt, only add the files that have something to do with the addition of the Bengali language  😉

**Reply by Iftakharpy** [reply]
Ok then I am done for now I will mark my PR as ready for review. Thank you so much for helping me.
